### PR TITLE
MM-20537 Migrate 'components/header_footer_template' module to Typescript.

### PR DESCRIPTION
**Pull request by openmohan**
#### Ticket Link

  Fixes https://github.com/mattermost/mattermost-server/issues/13704

### CODE REVIEW

File: components/header_footer_template/header_footer_template.test.tsx [hunk 1]
```diff
@@ -91,22 +90,24 @@ describe('components/HeaderFooterTemplate', () => {
...
         const wrapper = shallow(
             <NotLoggedIn config={{AboutLink: 'http://testaboutlink'}}/>,
         );
         expect(document.body.classList.contains('sticky')).toBe(true);
-        expect(document.getElementById('root').classList.contains('container-fluid')).toBe(true);
+        expect(rootElement && rootElement.classList.contains('container-fluid')).toBe(true);
         wrapper.unmount();
         expect(document.body.classList.contains('sticky')).toBe(false);
-        expect(document.getElementById('root').classList.contains('container-fluid')).toBe(false);
+        expect(rootElement && rootElement.classList.contains('container-fluid')).toBe(false);
```
Comment (reflog) [inline components/header_footer_template/header_footer_template.test.tsx:111]: you can shorten this to: 
```suggestion
        expect(rootElement?.classList?.contains('container-fluid')).toBe(false);
```
(i think :) )

**Comment by hanzei** [comment]
I'm deferring my review to someone more experienced with ts.

**Comment by lindy65** [comment]
Hi @reflog - can you help with test steps for this PR please?

**Comment by reflog** [comment]
@lindy65 - this is a migration ticket, nothing to QA here :) thanks!
